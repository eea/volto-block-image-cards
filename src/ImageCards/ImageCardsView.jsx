import React from 'react';
import RoundTiled from './RoundTiled';
import CardsGrid from './CardsGrid';
import Carousel from './Carousel';

const byDisplayType = {
  carousel: Carousel,
  round_tiled: RoundTiled,
  cards_grid: CardsGrid,
};

const ImageCardView = (props) => {
  const Impl = byDisplayType[props.data.display || 'carousel'];
  return <Impl {...props} />;
};

export default ImageCardView;
