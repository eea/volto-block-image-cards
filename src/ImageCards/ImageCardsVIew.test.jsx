import React from 'react';
import renderer from 'react-test-renderer';
import config from '@plone/volto/registry';
import ImageCardsView from './ImageCardsView';

config.blocks.blocksConfig = {
  imagecards: {
    blockRenderers: {
      id1: {
        view: () => <div>id1</div>,
      },
      id2: {
        view: () => <div>id2</div>,
      },
      id3: {
        view: () => <div>id3</div>,
      },
    },
  },
};

describe('ImageCardsView', () => {
  it('should render without crashing', () => {
    const component = renderer.create(
      <ImageCardsView data={{ display: 'id1' }} />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render without crashing', () => {
    const component = renderer.create(
      <ImageCardsView data={{ display: '' }} />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
