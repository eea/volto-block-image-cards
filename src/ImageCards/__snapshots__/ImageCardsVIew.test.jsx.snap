// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCardsView should render without crashing 1`] = `
<div
  className="image-cards-wrapper"
>
  <div>
    id1
  </div>
</div>
`;

exports[`ImageCardsView should render without crashing 2`] = `
<div
  className="image-cards-wrapper"
>
  
</div>
`;
